### ISSUE TITLE
USS Rename - File Opening flow

### ISSUE THREAD

**Issue by stepanzharychevbroadcom** [zowe/vscode-extension-for-zowe#563]
Here I will try to describe few cases which should be properly covered.

**1. Renaming opened file**
- User opens file;
- User renames file with any name;
- After successful renaming current tab is closed and temporary file related to the old filename is deleted;
- File is reopened;

**2. Renaming closed file**
- User renames file with any name;
- After successful renaming temporary file related to the old filename is deleted;
- File keeps it's closed state;

____
For **(1)** we already have function which ensures that the file is currently opened, so I see no problems to detect if we should reopen or not. In the same place we should resolve issues with removed state of node which happens immediately after rename.

**Comment by stepanzharychevbroadcom** [comment]
@Colin-Stone After checking of VSCode APIs I found out that there's no proper way to check if **binary** file is opened. Though we can check it for **text** file. So I would suggest the following way, we do smart opening for text files, but always open binary files. I guess it's not a big problem because binary files are less common case.

**Comment by Colin-Stone** [comment]
Agree with text file approach I wouldn't want to open a binary file if it was massive. So my feeling is that we could check if it is already downloaded and existing in local storage. If it is I would say download it. If not don't download it

**Comment by stepanzharychevbroadcom** [comment]
@Colin-Stone @zFernand0 Comment above is taken into account. Implementation is available here: https://github.com/zowe/vscode-extension-for-zowe/pull/576
Also I implemented separate function to ensure that text file is really opened in workspace, because I found out that **textDocuments** collection is not updated properly according to visibility state. Approach may look strange but that's what some extensions do due to poor VS Code API nature.
